runtime/src/bank/check_transactions.rs

```diff
@@ -202,8 +203,9 @@ impl Bank {
                 }
 
                 lock_result
-            })
-            .collect()
```

Comment:
we previously had 2 exact sized iterators so this collect on the zip should have only been doing a single allocation.